instrumentation.go

```diff
@@ -65,64 +62,62 @@ func NewInstrumentation(opts ...InstrumentationOption) (*Instrumentation, error)
 	}
 
 	c := newInstConfig(opts)
-	if err := c.validate(); err != nil {
-		return nil, err
-	}
-
-	pa := process.NewAnalyzer()
-	pid, err := pa.DiscoverProcessID(c.target)
-	if err != nil {
-		return nil, err
-	}
-
-	ctrl, err := opentelemetry.NewController(Version(), c.serviceName)
-	if err != nil {
-		return nil, err
-	}
 
-	mngr, err := instrumentors.NewManager(ctrl)
+	ctx := contextWithSigterm(context.Background())
```

Comment:
The change in context handling seems incorrect to me. We want users to still be able to cancel a `Run` meaning they need to be able to pass a context to that method.

Reply:
👍  Added context to `Run` method

Reply:
The context added to `Run` is still this context. It should not be. The creator of an `Instrumentor` needs to be the one creating and passing this context.

Reply:
Added it as arg.